demos/text-generation/llm.js

```diff
@@ -245,13 +245,32 @@ export class LLM {
 
         let attnMask = Array.from({ length: inputIdsLen }, () => BigInt(1));
         const positionIds = Array.from({ length: inputIdsLen }, (_, i) => BigInt(i++));
-        // Padding input_ids, attention_mask, position_ids to have length of this.maxLength
-        const inputIdsBuffer = this.paddingInput(inputIds, this.maxLength);
+        // Both input_ids and position_ids have shapes of [batch_size, sequence_length].
+        // the sequence_length is the length of inputIds, which is dynamic.
+        // Since WebNN does not support dynamic shapes, fix the sequence_length to maxLength and
+        // pad the rest elements with 0 value.
+        // TODO: This may cause an overflow error if maxLength is excessively large,
+        // as it could exceed the allowable array size or memory limits.
+        // e.g. QWen2.0 supports max_length: 32768, in a matmul of the GQA decomposed op,
+        // its input shapes will be [1, 14, 32768, 64] x [1, 14, 64, 32768] = [1, 14, 32768, 32768]
+        // which exceeds the 2GB tensor size limitation.
```

Comment:
Oof, entire models used to be < 100MB's, and now single tensors are passing the limits of 32-bit. It's no wonder that many ML libraries went with int64 for their indices. Related: https://github.com/microsoft/onnxruntime/issues/25445

> This happened in [chatGLM's](https://huggingface.co/THUDM/chatglm3-6b/blob/main/modeling_chatglm.py) mlp layer which has a weight martrix 4096 * 27392, then with batch size 256, the tensor elements will be 4096 * 27392 * 256 = 28,722,593,792 > 4,294,967,295.

(resolve me - no action except to read)